astacus/coordinator/plugins/clickhouse/steps.py

```diff
@@ -627,6 +653,22 @@ async def run_step(self, cluster: Cluster, context: StepsContext) -> None:
                     pass
 
 
+@dataclasses.dataclass
+class RestoreUDFsStep(Step[None]):
+    zookeeper_client: ZooKeeperClient
+    replicated_user_defined_zookeeper_path: str | None
+
+    async def run_step(self, cluster: Cluster, context: StepsContext) -> None:
+        if self.replicated_user_defined_zookeeper_path is None:
+            return
+
+        clickhouse_manifest = context.get_result(ClickHouseManifestStep)
+        async with self.zookeeper_client.connect() as connection:
+            for udf in clickhouse_manifest.udfs:
+                path = os.path.join(self.replicated_user_defined_zookeeper_path, udf.path)
+                await connection.create(path, udf.value)
```

Comment:
The tests using `wait_for_udf` suggests that when the restore is done, it's not actually done, it should be done.

The step itself should wait for the functions to be available before finishing the step.

Reply:
Good point